src/server/api/search.ts

```diff
@@ -48,27 +48,66 @@ export async function fetchResults(
   // Get the following in parallel:
   //  - search results
   //  - aggregations
-  const client = await nyplApiClient()
-  const [resultsResponse, aggregationsResponse] = await Promise.allSettled([
-    client.get(`${DISCOVERY_API_SEARCH_ROUTE}${resultsQuery}`),
-    client.get(`${DISCOVERY_API_SEARCH_ROUTE}${aggregationQuery}`),
-  ])
+  try {
+    // Failure to build client will throw from this:
+    const client = await nyplApiClient()
 
-  // Assign results values for each response when status is fulfilled
-  const results =
-    resultsResponse.status === "fulfilled" && resultsResponse.value
+    const [resultsResponse, aggregationsResponse] = await Promise.allSettled([
+      client.get(`${DISCOVERY_API_SEARCH_ROUTE}${resultsQuery}`),
+      client.get(`${DISCOVERY_API_SEARCH_ROUTE}${aggregationQuery}`),
+    ])
 
-  const aggregations =
-    aggregationsResponse.status === "fulfilled" && aggregationsResponse.value
+    // Handle failed promises (500)
+    if (
+      resultsResponse.status !== "fulfilled" ||
+      aggregationsResponse.status !== "fulfilled"
+    ) {
+      if (resultsResponse.status === "rejected") {
+        logServerError("fetchResults", resultsResponse.reason)
+        return {
+          status: 500,
+          message:
+            resultsResponse.reason instanceof Error
+              ? resultsResponse.reason.message
+              : resultsResponse.reason,
+        }
+      }
+      if (aggregationsResponse.status === "rejected") {
+        logServerError("fetchResults", aggregationsResponse.reason)
+        return {
+          status: 500,
+          message:
+            aggregationsResponse.reason instanceof Error
+              ? aggregationsResponse.reason.message
+              : aggregationsResponse.reason,
+        }
+      }
+    }
+
+    // Assign results values for each response
+    const results = resultsResponse.value
+
+    const aggregations = aggregationsResponse.value
+
+    // Handle invalid parameter rejection or empty results (422, 404)
+    if (
+      results.status === 422 ||
```

Comment:
are these actual error codes that are returned from the discovery api?

Reply:
If we are getting 4xx errors from discovery api, that indicates malformed queries made by RC. let's log these so we can keep tabs if these errors ever do turn up.